test/annexB/language/statements/for-in/nonstrict-initializer.js

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-initializers-in-forin-statement-heads
+description: >
+    for-in initializers in nonstrict mode
+flags: [noStrict]
+---*/
+
+(function() {
+  var effects = 0;
+  for (var a = ++effects in {});
+  assert.sameValue(effects, 1);
+})();
+
+
+(function() {
+  var stored;
+  for (var a = 0 in (stored = a, {}));
```

Comment:
The For-In statement supports an Expression here, so the parenthesis aren't
required. Removing them might make this a little more useful as a syntax
test--ensuring that there aren't inconsistent grammar restrictions in the
presence of the initializer.